app/models/iiif_canvas_presenter.rb

```diff
@@ -96,225 +96,232 @@ def service
-        transcript_url = Rails.application.routes.url_helpers.transcripts_master_file_supplemental_file_url(master_file.id, file.id)
-        [IIIFManifest::V3::AnnotationContent.new(body_id: caption_url, **supplemental_attributes(file, type: 'caption')),
-         IIIFManifest::V3::AnnotationContent.new(body_id: transcript_url, **supplemental_attributes(file, type: 'transcript'))]
-      else
-        url = Rails.application.routes.url_helpers.master_file_supplemental_file_url(master_file.id, file.id)
-        IIIFManifest::V3::AnnotationContent.new(body_id: url, **supplemental_attributes(file))
-      end
+  def supplementing_content_data(file)
+    tags = file.tags.reject { |t| t == 'machine_generated' }.compact
+    case tags
+    when ['caption']
+      url = Rails.application.routes.url_helpers.captions_master_file_supplemental_file_url(master_file.id, file.id)
+      IIIFManifest::V3::AnnotationContent.new(body_id: url, **supplemental_attributes(file, type: 'caption'))
+    when ['transcript']
+      url = Rails.application.routes.url_helpers.transcripts_master_file_supplemental_file_url(master_file.id, file.id)
+      IIIFManifest::V3::AnnotationContent.new(body_id: url, **supplemental_attributes(file, type: 'transcript'))
+    when ['description']
+      url = Rails.application.routes.url_helpers.descriptions_master_file_supplemental_file_url(master_file.id, file.id)
+      IIIFManifest::V3::AnnotationContent.new(body_id: url, **supplemental_attributes(file, type: 'description'))
+    when ['caption', 'transcript']
+      caption_url = Rails.application.routes.url_helpers.captions_master_file_supplemental_file_url(master_file.id, file.id)
+      transcript_url = Rails.application.routes.url_helpers.transcripts_master_file_supplemental_file_url(master_file.id, file.id)
+      [IIIFManifest::V3::AnnotationContent.new(body_id: caption_url, **supplemental_attributes(file, type: 'caption')),
+       IIIFManifest::V3::AnnotationContent.new(body_id: transcript_url, **supplemental_attributes(file, type: 'transcript'))]
+    else
+      url = Rails.application.routes.url_helpers.master_file_supplemental_file_url(master_file.id, file.id)
+      IIIFManifest::V3::AnnotationContent.new(body_id: url, **supplemental_attributes(file))
     end
+  end
 
-    def stream_urls
-      stream_info[:stream_hls].collect do |d|
-        [d[:quality], d[:url], d[:mimetype]]
-      end
+  def stream_urls
+    stream_info[:stream_hls].collect do |d|
+      [d[:quality], d[:url], d[:mimetype]]
     end
+  end
 
-    def section_processing?(master_file)
-      !master_file.succeeded?
-    end
+  def section_processing?(master_file)
+    !master_file.succeeded?
+  end
 
-    def supplemental_captions_transcripts
-      master_file.supplemental_files(tag: 'caption') + master_file.supplemental_files(tag: 'transcript')
+  def supplemental_accessibility_files
```

Comment:
I wonder if we could remove the need for this method by extending `MasterFiles#supplemental_files` to take an array for the `tag` kwarg.